pkgs/native_assets_cli/test/api/build_config_test.dart

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:cli_config/cli_config.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_assets_cli/native_assets_cli_internal.dart' as internal;
+import 'package:test/test.dart';
+
+void main() async {
+  late Uri tempUri;
+  late Uri outDirUri;
+  late Uri outDir2Uri;
+  late String packageName;
+  late Uri packageRootUri;
+  late Uri fakeClang;
+  late Uri fakeLd;
+  late Uri fakeAr;
+  late Uri fakeCl;
+  late Uri fakeVcVars;
+
+  setUp(() async {
+    tempUri = (await Directory.systemTemp.createTemp()).uri;
+    outDirUri = tempUri.resolve('out1/');
+    await Directory.fromUri(outDirUri).create();
+    outDir2Uri = tempUri.resolve('out2/');
+    packageName = 'my_package';
+    await Directory.fromUri(outDir2Uri).create();
+    packageRootUri = tempUri.resolve('$packageName/');
+    await Directory.fromUri(packageRootUri).create();
+    fakeClang = tempUri.resolve('fake_clang');
+    await File.fromUri(fakeClang).create();
+    fakeLd = tempUri.resolve('fake_ld');
+    await File.fromUri(fakeLd).create();
+    fakeAr = tempUri.resolve('fake_ar');
+    await File.fromUri(fakeAr).create();
+    fakeCl = tempUri.resolve('cl.exe');
+    await File.fromUri(fakeCl).create();
+    fakeVcVars = tempUri.resolve('vcvarsall.bat');
+    await File.fromUri(fakeVcVars).create();
+  });
+
+  tearDown(() async {
+    await Directory.fromUri(tempUri).delete(recursive: true);
+  });
+
+  test('BuildConfig ==', () {
+    final config1 = BuildConfig(
+      outDir: outDirUri,
+      packageName: packageName,
+      packageRoot: tempUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.iOS,
+      targetIOSSdk: IOSSdk.iPhoneOs,
+      cCompiler: CCompilerConfig(
+        cc: fakeClang,
+        ld: fakeLd,
+        ar: fakeAr,
+      ),
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+    );
+
+    final config2 = BuildConfig(
+      outDir: outDir2Uri,
+      packageName: packageName,
+      packageRoot: tempUri,
+      targetArchitecture: Architecture.arm64,
+      targetOs: OS.android,
+      targetAndroidNdkApi: 30,
+      buildMode: BuildMode.release,
+      linkModePreference: LinkModePreference.preferStatic,
+    );
+
+    expect(config1, equals(config1));
```

Comment:
Why test this?

Reply:
Checking that we didn't mess up the `operator ==` implementation.